### DISCUSSION TITLE
Whiten/Darken Deprecation

### DISCUSSION THREAD

**Discussion by real-lazy-coder** [chakra-ui/chakra-ui#6139] [General] [answered]
Will there be any alternatives to these utility functions?  

```ts
import { darken, whiten } from "@chakra-ui/theme-tools"
```

**Comment by arnoldsandoval** [comment]
@real-lazy-coder for my knowledge, where did you see these would be deprecated?

**Reply by real-lazy-coder** [reply]
The typescript typing's have it depricated.

![image](https://user-images.githubusercontent.com/6243027/171857674-3cc05a44-11d6-46fb-b89a-d8d71c608f00.png)

**Comment by bbcntolga** [comment]
![Screenshot 2022-06-12 at 12 37 10](https://user-images.githubusercontent.com/51093354/173229236-68b7c605-0904-4606-bee8-b44895951f2f.png)

The transparentize is also tagged as deprecated. Is there any new information about this? Any alternatives?

**Comment by JTBrinkmann** [comment, marked as answer]
I don't think it was ever advertised as part of the public API. If it's not used internally anymore, I think it makes sense to deprecate and remove it. As these are rather small functions, you could copy them into your own projects when they get removed.

**Reply by JTBrinkmann** [reply]
when I say "copy them", I mean including a color manipulation library to do the heavy lifting. It currently uses [@ctrl/tinycolor](https://www.npmjs.com/package/@ctrl/tinycolor). With minor code changes, it might be possible to reduce the bundle size using [simple-color-functions](https://www.npmjs.com/package/simple-color-functions) instead.

**Comment by noobinthisgame** [comment]
As @JTBrinkmann mentioned, we are not using most of those functions internally and thus will remove them. This would enable us to remove the dependency `@crtl/tinycolor` and reduce our bundle size. 
We don't plan on providing a replacement, if you need such function in your project you can use your library of choice in your custom theme :)

**Reply by SeunggiLeeUSM** [reply]
Hello, one thing I want to point out about this whole deprecation issue is, many of these functions are currently used in some of the chakra-ui pro components codes.
For example, in the Color picker component provided in this page https://pro.chakra-ui.com/components/e-commerce/product-pickers, you will see the isDark() and transparentize() currently being used in the code provided. So maybe need to update the chakra pro codes if we are deprecating many of those helper functions.

**Reply by bragle** [reply]
It's not only used in the components, but also the [themes](https://pro.chakra-ui.com/themes). Would be nice if these could be updated before deprecating the dependencies.

**Reply by la55u** [reply]
Not only the pro themes but the open source ones are [using it too ](https://github.com/chakra-ui/chakra-ui/blob/9e3d91fd73b1cadc14f98b6c834c0e6faf134bd2/packages/components/theme/src/components/button.ts#L39) and these themes are linked directly on the docs site so I'd argue they are publicly advertised. It would be nice to see an alternative implementation before they get removed.
